cmake/external/icu.cmake

```diff
@@ -0,0 +1,170 @@
+    
+    # Calculate the number of cores using CMake
+    execute_process(COMMAND nproc OUTPUT_VARIABLE CMAKE_JOB_POOL_SIZE)
+    string(STRIP ${CMAKE_JOB_POOL_SIZE} CMAKE_JOB_POOL_SIZE)
+endif()
+
+set(ICU_INCLUDE_DIRS "${ICU_INSTALL_DIR}/include")
+set(ICU_STATIC_LIB_DIR "${ICU_INSTALL_DIR}/${ICU_INSTALL_LIB_SUBDIR}")
+set(ICU_SHARED_LIB_DIR "${ICU_INSTALL_DIR}/${ICU_INSTALL_BIN_SUBDIR}")
+
+set(ICU_UC_LIB "${ICU_STATIC_LIB_DIR}/${ICU_STATIC_PREFIX}${ICU_UC_LIB_NAME}${ICU_SUFFIX}${ICU_STATIC_SUFFIX}")
+set(ICU_I18N_LIB "${ICU_STATIC_LIB_DIR}/${ICU_STATIC_PREFIX}${ICU_I18N_LIB_NAME}${ICU_SUFFIX}${ICU_STATIC_SUFFIX}")
+set(ICU_DATA_LIB "${ICU_STATIC_LIB_DIR}/${ICU_STATIC_PREFIX}${ICU_DATA_LIB_NAME}${ICU_STATIC_SUFFIX}")
+
+set(ICU_UC_SHARED_LIB "${ICU_SHARED_LIB_DIR}/${ICU_SHARED_PREFIX}${ICU_UC_SHARED_LIB_NAME}${ICU_SUFFIX}${ICU_SHARED_SUFFIX}")
+set(ICU_I18N_SHARED_LIB "${ICU_SHARED_LIB_DIR}/${ICU_SHARED_PREFIX}${ICU_I18N_SHARED_LIB_NAME}${ICU_SUFFIX}${ICU_SHARED_SUFFIX}")
+set(ICU_DATA_SHARED_LIB "${ICU_SHARED_LIB_DIR}/${ICU_SHARED_PREFIX}${ICU_DATA_SHARED_LIB_NAME}${ICU_SHARED_SUFFIX}")
+
+set(ICU_LIBRARIES ${ICU_UC_LIB} ${ICU_I18N_LIB} ${ICU_DATA_LIB})
+
+if(WIN32)
+  set(ICU_SHARED_LIBRARIES ${ICU_UC_SHARED_LIB} ${ICU_I18N_SHARED_LIB} ${ICU_DATA_SHARED_LIB})
+endif()
+
+include(ExternalProject)
+
+if(WIN32)
+  ExternalProject_Add(
+    ${ICU_TARGET_NAME}
+    URL ${ICU_URL}
+    URL_HASH ${ICU_URL_HASH}
+    PREFIX ${THIRD_PARTY_PATH}
+    SOURCE_DIR ${ICU_SOURCE_DIR}
+    INSTALL_DIR ${ICU_INSTALL_DIR}
+    CONFIGURE_COMMAND msbuild ${ICU_SOURCE_DIR}\\source\\allinone\\allinone.sln /p:Configuration=${ICU_BUILD_TYPE} /p:Platform=x64 /t:common /t:i18n /t:uconv /t:makedata 
+    BUILD_COMMAND ""
+    INSTALL_COMMAND ${CMAKE_COMMAND} -E copy_directory ${ICU_SOURCE_DIR}/include ${ICU_INSTALL_DIR}/include && 
+                    ${CMAKE_COMMAND} -E copy_directory ${ICU_SOURCE_DIR}/lib64 ${ICU_INSTALL_DIR}/lib64 &&
+                    ${CMAKE_COMMAND} -E copy_directory ${ICU_SOURCE_DIR}/bin64 ${ICU_INSTALL_DIR}/bin64
+    DOWNLOAD_EXTRACT_TIMESTAMP ON
+    BUILD_BYPRODUCTS ${ICU_LIBRARIES}
+  )
+elseif(APPLE)
+  ExternalProject_Add(
+    ${ICU_TARGET_NAME}
+    URL ${ICU_URL}
+    URL_HASH ${ICU_URL_HASH}
+    PREFIX ${THIRD_PARTY_PATH}
+    SOURCE_DIR ${ICU_SOURCE_DIR}
+    INSTALL_DIR ${ICU_INSTALL_DIR}
+    CONFIGURE_COMMAND ${HOST_ENV_CMAKE} ${ICU_SOURCE_DIR}/source/runConfigureICU MacOSX --prefix ${ICU_INSTALL_DIR}
+                      ${ICU_CONFIGURE_FLAGS} 
+                      --enable-static
+                      --enable-rpath
+                      --disable-shared
+                      --disable-tests
+                      --disable-samples
+                      --disable-extras
+                      --disable-icuio
+                      --disable-draft
+                      --disable-icu-config
+    BUILD_COMMAND make -j${CMAKE_JOB_POOL_SIZE} 
+    INSTALL_COMMAND make install
+    DOWNLOAD_EXTRACT_TIMESTAMP ON
```

Comment:
it's not available for older versions of cmake https://cmake.org/cmake/help/latest/policy/CMP0135.html